### PR TITLE
week6 version 2

### PR DESCRIPTION
**Pull request by nk9487**

### CODE REVIEW

File: service/app.js [hunk 1]
```diff
@@ -0,0 +1,282 @@
...
+app.use(exampleMiddleware)
+
+// Get  Routes
+
+app.get('/', (req, res) => {
+  res.send('Hello, world!')
+})
+
+app.get('/decks', (req, res) => {
+  res.sendStatus(404)
```
Comment (panzerama) [inline service/app.js]: What is the purpose of this route?

File: service/app.js [hunk 2]
```diff
@@ -0,0 +1,282 @@
...
+})
+//==============================Get Deck by ID ==============================//
+
+app.get('/decks/:id/cards', async (req, res) => {
+  console.log('request id ', req.params.id)
```
Comment (panzerama) [inline service/app.js]: Don't leave unnecessary log statements in finished code.

File: service/app.js [hunk 3]
```diff
@@ -0,0 +1,282 @@
...
+  const limit = req.query.limit
+  console.log('query param limit ', limit)
+  const deck = await Deck.findById(req.params.id)
+  if (deck){
+    console.log('found card', deck.cards.length)
+   res.send(deck.cards.slice(0,3))
```
Comment (panzerama) [inline service/app.js:44]: Where is "3" coming from?

File: service/app.js [hunk 4]
```diff
@@ -0,0 +1,282 @@
...
+
+  res.status(200).send(card)
+}
+app.get('/cards/:id', cardsById)
+
+//==============================Get A Deck By A User Id==============================//
+
+app.get('/decks/:userId/deckCards', async (req, res) => {
+  console.log('user ID ', req.params.userId)
+  //const limit = req.query.limit
```
Comment (panzerama) [inline service/app.js]: Unnecessary comments

File: service/app.js [hunk 5]
```diff
@@ -0,0 +1,282 @@
...
+    'cards._id': req.params.id
+  })
+
+  res.status(200).send(card)
+}
+app.get('/cards/:id', cardsById)
+
+//==============================Get A Deck By A User Id==============================//
+
+app.get('/decks/:userId/deckCards', async (req, res) => {
```
Comment (panzerama) [inline service/app.js:61]: This naming is confusing, and one of the core principles behind a REST API is that it should be discoverable. What other ways can you think of to write this path?

File: service/app.js [hunk 6]
```diff
@@ -0,0 +1,282 @@
...
+  }
+})
+
+//==========================================Create Card============================================================================
+const isUrl = (value) => {
+  const re = /((([A-Za-z]{3,9}:(?:\/\/)?)(?:[-;:&=\+\$,\w]+@)?[A-Za-z0-9.-]+|(?:www.|[-;:&=\+\$,\w]+@)[A-Za-z0-9.-]+)((?:\/[\+~%\/.\w-_]*)?\??(?:[-\+=&;%@.\w_]*)#?(?:[\w]*))?)/
+  return re.test(value)
+}
+
+app.post('/createCards', async (req, res) => {
```
Comment (panzerama) [inline service/app.js]: When we're writing a new route, we should use nouns in the path to represent the resource we're making changes to and let the HTTP methods suggest the action. the verb here is POST and the noun is cards, so we don't need 'create'

File: service/app.js [hunk 7]
```diff
@@ -0,0 +1,282 @@
...
+      return res.sendStatus(200)
+    } catch (error) {
+      console.log("error creating decks")
+    }
+
+})
+
+//===============================================create user===========================================================================
+//i beliieve this wont be pushed since we have not created the user on mongodb
+app.post("/createUser", async (req, res) => {
```
Comment (panzerama) [inline service/app.js:139]: same note here about paths

File: service/app.js [hunk 8]
```diff
@@ -0,0 +1,282 @@
...
+      const result = await deck.save()
+      return res.sendStatus(200)
+    } catch (error) {
+      console.log("error creating decks")
+    }
+
+})
+
+//===============================================create user===========================================================================
+//i beliieve this wont be pushed since we have not created the user on mongodb
```
Comment (panzerama) [inline service/app.js:138]: Why wouldn't you create test data for Users in MongoDB?

File: service/app.js [hunk 9]
```diff
@@ -0,0 +1,282 @@
...
+  }
+})
+
+//=======================================update card=================================================================================
+
+app.put('/updatecard/:Id', async (req, res) => {
+  const cardRequest = req.body
+
+  try {
+    const deck = await Deck.findById(cardRequest.deckId)
```
Comment (panzerama) [inline service/app.js:184]: Passing the deck id in the request is one way to accomplish this, but it is a departure from the other routes we've designed. Think about how you might pass the deck id in the path.

File: service/app.js (hunk 3 above)
Comment (nk9487) [reply service/app.js:44]: It was manually typed, I think it should have been getting this as input.

File: service/app.js (hunk 2 above)
Comment (nk9487) [reply service/app.js]: got it, removed.

File: service/app.js (hunk 1 above)
Comment (nk9487) [reply service/app.js]: These was unnecessary route, it is removed.

File: service/app.js (hunk 4 above)
Comment (nk9487) [reply service/app.js]: removed.

File: service/app.js (hunk 5 above)
Comment (nk9487) [reply service/app.js:61]: I think  '/decks/:userId' would be enough.

File: service/app.js (hunk 6 above)
Comment (nk9487) [reply service/app.js]: ok, it makes sense.

**Pull request merged**